client/client-lib/src/mint/backup.rs

```diff
@@ -259,18 +259,16 @@ impl MintClient {
         backup: PlaintextEcashBackup,
         gap_limit: usize,
     ) -> Result<Cancellable<EcashRecoveryFinalState>> {
-        let end_epoch = match self.context.api.fetch_last_epoch().await {
+        let epoch_count = match self.context.api.fetch_epoch_count().await {
             Ok(v) => v,
             Err(e) => {
                 return Err(e.into());
             }
         };
-        let epoch_range = backup.epoch..=end_epoch;
+        let start_epoch = backup.epoch_count.saturating_sub(1);
+        let epoch_range = start_epoch..epoch_count;
```

Comment:
This reads confusingly now, IMO. 

```
let epoch_range = backup.epoch_count.saturating_sub(1)..current_epoch_count;
```

would be best, I guess. We might need to rename the field in the `PlaintextEcashBackup`, but I think that's a good idea anyway, as it is more precise.

Reply:
BTW, that `saturating_sub(1)` is probably unnecessary, but I added it because I didn't have confidence this is the case. I should have probably added a comment.

Can you please add `// TODO: This -1 is probably not neccessary, as it should be enough to start from the exact epoch the snapshot was taken, but it is harmless to start from any epoch in the past.`. Thanks.

Reply:
Thanks for your comments.

Regarding the `-1`, if I understand correctly the backup contains `backup.epoch_count` epochs, with the last backup epoch being `backup.epoch_count - 1`, and so with the `-1` we'll start from the same epoch the backup was taken. If I'm reading your suggested comment correctly, it implies that by using `-1` we'll start in an earlier epoch than when the backup was taken, which I don't think is the case. On a related note, could we also start from the next epoch after the backup was taken, so without the `-1`?

Another thought I had was whether it makes sense to exit early in case the backup and current `epoch_count` are the same. Even though `epoch_range` would be empty, we might do some extra steps before we realize.

Regarding removing the `start_epoch` variable, I'm currently using it in the `info!` call below, which I could also replace with `backup.epoch_count.saturating_sub(1)`, but I thought it might be better to avoid repeating that.

Sorry, I hope I'm not overcomplicating this, but just wanted to make sure :)

Reply:
The backup of the notes from the database is taken at the epoch count N, which means any changes to note state could have only happen in the next epoch (so epoch N), so starting from just N (instead of N - 1) *should* be OK. I wast just not sure if they weren't any weird races or corner cases.

> Another thought I had was whether it makes sense to exit early in case the backup and current epoch_count are the same.

Probability of this is almost zero in practice, so not worth special-casing. Generally decent logic should not need any `if something.is_empty() { /* early exit */ return; }`. Rather common anti pattern.

> Regarding removing the start_epoch variable, I'm currently using it in the info! call below, which I could also replace with backup.epoch_count.saturating_sub(1), but I thought it might be better to avoid repeating that.

I guess I only care about `epoch_count` being named `current_epoch_count`.

Reply:
Ah yes I see, thanks for clarifying about the backup epoch. I've now added your comment to the code.

> Probability of this is almost zero in practice, so not worth special-casing.

OK thanks for confirming, and also thanks for the tip about the anti-pattern :)

> I guess I only care about `epoch_count` being named `current_epoch_count`.

Ah sorry, I hadn't spotted this renaming in your previous comment. Have now made this change and left `start_epoch` unchanged.